### PR TITLE
Feat: Add Promql aggregation topk, bottomk

### PR DESCRIPTION
**Pull request by Jiale-Fang**
# Description
Add Promql aggregation topk, bottomk

# Testing
Describe how you tested this code. How can the reviewers reproduce your tests?

# Checklist:
Before marking your pull request as ready for review, complete the following.

- [x] I have self-reviewed this PR.
- [x] I have removed all print-debugging and commented-out code that should not be merged.
- [x] I have added sufficient comments in my code, particularly in hard-to-understand areas.
- [x] I have formatted the code, if applicable. For Go, I have run `goimports -w .`.

### CODE REVIEW

**Comment by nkunal** [comment]
@Jiale-Fang can you resolve the merge conflicts on this PR

File: pkg/segment/results/mresults/metricresults.go [hunk 1]
```diff
@@ -702,14 +703,83 @@ func CalculateInterval(timerangeSeconds uint32) (uint32, error) {
...
+	}
+
+	for timestamp, pq := range timestampToHeap {
+		for pq.Len() > 0 {
+			item := heap.Pop(pq).(*utils.Item)
+			_, exists := r.Results[item.Value]
+			if !exists {
+				r.Results[item.Value] = make(map[uint32]float64)
+			}
+			r.Results[item.Value][timestamp] = (item.Priority * factor) // Restore to original value
```
Comment (AndrewHess) [inline pkg/segment/results/mresults/metricresults.go:776]: I think you actually want
```
(item.Priority / factor) // Restore to original value
```
Right? In your usage you only use `factor = 1` or `factor = -1`, so `*` and `/` are equivalent, but for a more general `factor` you should use division

File: pkg/segment/results/mresults/metricresults.go [hunk 2]
```diff
@@ -702,14 +703,83 @@ func CalculateInterval(timerangeSeconds uint32) (uint32, error) {
...
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// The larger the priority, the earlier it will be popped out. Since we use the value as the priority, for `topk`, the larger the value, the more we want it to remain in the priority queue. Therefore, its priority should be smaller.
+// For bottomk, it's the opposite
+func (r *MetricsResult) computeExtremesKElements(funcConstant float64, factor float64, isTopK bool) error {
```
Comment (AndrewHess) [inline pkg/segment/results/mresults/metricresults.go:727]: I think `factor` and `isTopK` are acting as redundant pieces of information. Do you only need to have one of them as a parameter?

Comment (Jiale-Fang) [reply pkg/segment/results/mresults/metricresults.go:727]: I need `isTopK` to check these 2 conditions. And `factor` can help us to generate priority for each pq item without using if statement. What do you think?
```

	if isTopK && (item.Priority*factor) > runningDS.runningEntries[i].runningVal {
		heap.Push(pq, item)
		continue
	}
	if !isTopK && (item.Priority) < runningDS.runningEntries[i].runningVal {
		heap.Push(pq, item)
		continue
	}
```

**Review by AndrewHess — APPROVED** [review]

**Pull request merged**